### PR TITLE
support function2.0 in project folder

### PR DESCRIPTION
**Pull request by watashiSHUN**
if the XYZ.csproj contains
`<PackageReference Include="Microsoft.NET.Sdk.Functions"  />`
we view it as a msbuild template candidate

otherwise we assume its extensions.csproj
we view it as a basic template candidate

this should resolve https://github.com/projectkudu/kudu/issues/2631

### CODE REVIEW

File: Kudu.Core/Deployment/Generator/SiteBuilderFactory.cs [hunk 1]
```diff
@@ -331,12 +331,25 @@ private ISiteBuilder DetermineProject(string repositoryRoot, string targetPath,
             }
             else if (FunctionAppHelper.LooksLikeFunctionApp())
             {
-                return new FunctionMsbuildBuilder(_environment,
-                                                perDeploymentSettings,
-                                                _propertyProvider,
-                                                repositoryRoot,
-                                                targetPath,
-                                                solutionPath);
```
Comment (davidebbo) [inline Kudu.Core/Deployment/Generator/SiteBuilderFactory.cs]: Please avoid unrelated spacing changes to keep the diff minimal. Don't change it this time, but keep in mind for next time. Basically, if you see anything in the diff that doesn't directly related to your PR, it probably shouldn't be included :)

**Comment by watashiSHUN** [comment]
merged https://github.com/projectkudu/kudu/commit/5331507d682d31480ab52f3323692a35d7d7cd76
